srcalc/src/main/java/gov/va/med/srcalc/util/TabularUploadError.java

```diff
@@ -0,0 +1,192 @@
+package gov.va.med.srcalc.util;
+
+import java.util.*;
+
+import org.springframework.context.MessageSourceResolvable;
+import org.springframework.validation.DefaultMessageCodesResolver;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * Similar to Spring's {@link org.springframework.validation.FieldError} class,
+ * but specifically represents an error in a tabular file upload.
+ */
+public final class TabularUploadError implements MessageSourceResolvable
+{
+    /**
+     * Special row number value indicating a global error on the whole upload instead of
+     * a specific row.
+     */
+    public static int ROW_NUMBER_GLOBAL = -1;
+    
+    /**
+     * The object name used for errors on the whole upload.
+     */
+    public static String OBJECT_NAME = "tabularUpload";
+    
+    private static final DefaultMessageCodesResolver fCodesResolver =
+            new DefaultMessageCodesResolver();
+    
+    private final int fRowNumber;
+    private final String fColumnName;
+    private final String[] fCodes;
+    private final Object[] fArguments;
+    private final String fDefaultMessage;
+    
+    /**
+     * Constructs an instance.
+     * @param rowNumber the associated row's number (1-based). May also be {@link #ROW_NUMBER_GLOBAL}
+     * @param columnName the associated field name of the target object. Must not be null.
+     * @param codes the error codes. Must be a non-empty array.
+     * @param arguments any necessary message arguments. May be null.
+     * @param defaultMessage the default message to use if no codes could be resolved
+     * @throws IllegalArgumentException if codes is empty
+     */
+    public TabularUploadError(
+            final int rowNumber,
+            final String columnName,
+            final String[] codes,
+            final Object[] arguments,
+            final String defaultMessage)
+    {
+        fRowNumber = rowNumber;
+        fColumnName = Objects.requireNonNull(columnName, "field must not be null");
+        fCodes = codes.clone();
+        if (fCodes.length < 1)
+        {
+            throw new IllegalArgumentException("codes must be non-empty");
+        }
+        fArguments = arguments;
+        fDefaultMessage = defaultMessage;
+    }
+    
+    /**
+     * Creates an error for the whole upload.
+     * @param code the error code. Must be non-null.
+     * @param arguments any necessary message arguments. May be null.
+     * @param defaultMessage the default message to use if no codes could be resolved
+     * @return a new instance
+     */
+    public static TabularUploadError global(
```

Comment:
I think it would help to describe what a global error is here. Maybe one or two examples would help.